### PR TITLE
Add delete concept flow

### PR DESCRIPTION
**Pull request by glokta1**
**Checklist**:

- [X] I've thoroughly self-reviewed my changes
- [X] I've added tests for my changes, unless they affect admin-only areas (or are otherwise not worth testing)
- [ ] I've verified any visual changes using Percy (add a commit with `[percy]` in the message to trigger)

## Summary by CodeRabbit

- **New Features**
  - Introduced a modal for deleting concepts with confirmation and timed button press.
  - Added a "Danger Zone" section in the concept admin interface for deletion.

- **Bug Fixes**
  - Addressed user authorization checks before allowing concept deletion.

- **Tests**
  - Implemented acceptance tests for concept deletion functionality.
  - Added helper functions for user authentication in tests.

- **Chores**
  - Updated route handlers and model attributes to support concept deletion.

- **Refactor**
  - Enhanced existing controllers and templates to integrate new deletion functionality.

### CODE REVIEW

**Comment by glokta1** [comment]
@rohitpaulk 
1. "Test Results" check seems to be failing due to changes not in this PR. 

2. Checked out the CodeCov report and it's complaining about the same two lines as the one in `delete-repository-modal.ts`. Those won't get run unless the route handler fails to load the model from the Store.

**Comment by rohitpaulk** [comment]
@glokta1 this is ready for review yes? (The review request was dismissed so thought I'd double check)

**Comment by glokta1** [comment]
@rohitpaulk Not yet. I realised I was writing logic for some things that already had helpers defined. Changing that + making the tests more comprehensive

File: app/components/concept-admin/delete-concept-modal.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  };
+};
+
+export default class DeleteConceptModalComponent extends Component<Signature> {
+  @service router!: RouterService;
+  @service store!: Store;
+
+  @action
+  async deleteConcept() {
+    if (!this.args.concept) {
```
Comment (rohitpaulk) [inline app/components/concept-admin/delete-concept-modal.ts:23]: When will this happen? The type for `concept` suggests that it'll always be `ConceptModel`.

File: app/controllers/concept-admin/basic-details.ts [hunk 2]
```diff
@@ -7,17 +7,33 @@ import { task, timeout } from 'ember-concurrency';
...
+import type UserModel from 'codecrafters-frontend/models/user';
+import type AuthenticatorService from 'codecrafters-frontend/services/authenticator';
 
 export default class BasicDetailsController extends Controller {
   @service declare store: Store;
   @service declare router: RouterService;
-
-  @tracked wasSavedRecently = false;
+  @service declare authenticator: AuthenticatorService;
+  @tracked wasSavedRecently: boolean = false;
```
Comment (rohitpaulk) [inline app/controllers/concept-admin/basic-details.ts:17]: Let's keep an empty line between services & tracked properties (like all other files do)

**Review by rohitpaulk — APPROVED** [review]

**Review by rohitpaulk — APPROVED** [review]

**Pull request merged**